dist/dfu.js

```diff
@@ -134,6 +134,9 @@
             })
             .then(function() {
                 log("modeData written");
+                setTimeout(function() {  // TODO: Remove this when gattserverdisconnected event is implemented and possibly put a timeout in that event handler before resolving.
```

Comment:
I've got a fix half done for the multiple resolve(), but I cant test it till later as my device isn't with me. I can try tonight, or perhaps you can get up and running with node and make a suggestion?